fix_includes_test.py

```diff
@@ -1884,9 +1885,76 @@ def testIncludeComments(self):
     self.RegisterFileContents({'subdir/include_comments.cc': infile})
     self.ProcessAndTest(iwyu_output)
 
+  def testIncludeNoComments(self):
+    """Tests that we don't include comments on #include lines. with --comments=none"""
+    self.flags.comments = 'none'
+    infile = """\
+// Copyright 2010
+
+#include <notused.h>  ///-
+///+#include "subdir/include_no_comments.h"
+///+#include <stdio.h>
+#include "used.h"
+///+#include "used2.h"
+
+int main() { return 0; }
+"""
+    iwyu_output = """\
+subdir/include_no_comments.cc should add these lines:
+#include "subdir/include_no_comments.h"
+#include <stdio.h>
+#include "used2.h"
+
+subdir/include_no_comments.cc should remove these lines:
+- #include <notused.h>  // lines 3-3
+
+The full include-list for subdir/include_no_comments.cc:
+#include "subdir/include_no_comments.h"
+#include <stdio.h>
+#include "used.h"
+#include "used2.h"
+---
+"""
+    self.RegisterFileContents({'subdir/include_no_comments.cc': infile})
+    self.ProcessAndTest(iwyu_output)
+
+  def testIncludeLongComments(self):
+    """Tests that we properly include long comments on #include lines."""
+    self.flags.comments='long'
+    infile = """\
+// Copyright 2010
+
+#include <notused.h>  ///-
+///+#include "subdir/include_long_comments.h"
+///+#include <stdio.h>   // for std::printf(), etc.
+#include "used.h"    ///-
+///+#include "used.h"    // for NS::Used
+///+#include "used2.h"   // for NS::Used2, NS::Used2::Used2, NS::Used2::~Used2, NS::Used2::Used_Enum, operator==()
+
+int main() { return 0; }
+"""
+    iwyu_output = """\
+subdir/include_long_comments.cc should add these lines:
+#include "subdir/include_long_comments.h"
+#include <stdio.h>   // for printf(), etc.
+#include "used2.h"   // for Used2, Used2::Used2, Used2::~Used2, Used2::Used_Enum, operator==()
+
+subdir/include_long_comments.cc should remove these lines:
+- #include <notused.h>  // lines 3-3
+
+The full include-list for subdir/include_long_comments.cc:
+#include "subdir/include_long_comments.h"
+#include <stdio.h>   // for std::printf(), etc.
+#include "used.h"    // for NS::Used
+#include "used2.h"   // for NS::Used2, NS::Used2::Used2, NS::Used2::~Used2, NS::Used2::Used_Enum, operator==()
+---
+"""
+    self.RegisterFileContents({'subdir/include_long_comments.cc': infile})
+    self.ProcessAndTest(iwyu_output)
+
   def testNocommentsFlag(self):
     """Tests we properly don't include/modify comments with --nocomments."""
-    self.flags.comments = False
+    self.flags.nocomments = True
```

Comment:
This test isn't very useful -- I don't think `nocomments` affects the execution at all (does it?).

So I suspect this just runs with the default value for `comments`.